### PR TITLE
Jbaker nested menus/#79

### LINKED ISSUE #79: Support for sub-menu UI
## Is your feature request related to a problem? Please describe.

The design spec illustrated an intended sub-menu option. 

## Describe the solution you'd like

Build option 

Screen Shot 2022-01-28 at 11 30 46 AM0

Exit Criteria:  The menu ui changes are addressed
https://github.com/AlaskaAirlines/aurolabs-suggest/issues/8

### LINKED ISSUE #98: Change hover state color and spacing
## Is your feature request related to a problem? Please describe.

The contrast between the default background and hover state is too low, making it hard to see. The space between the menu and check box is also too large.

## Describe the solution you'd like

1. Decrease spacing between checkbox and left side of menu
2. Change hover state background color to bkg-hover token

## Additional context

Refer to this blueprint with updated specs: https://www.figma.com/file/xwDiIDfSz6j9bM4TD3Ni5N/Menu?node-id=322%3A298

Development for this updated feature should be merged into the `v3.2-rc` branch, not directly to `main`. 

## Exit criteria 

This issue will be considered complete once the engineered solution is reviewed with @leeejune and is approved.

### PR DESCRIPTION
**Pull request by jason-capsule42**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Fixes:** #79, #98

## Summary:

This change makes it possible to directly nest one auro-menu inside of another.

Additionally, the selected, active and hover state colors for menuoptions were updated per discussion and review with @leeejune

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [ ] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: demo/demo.md [hunk 1]
```diff
@@ -90,6 +90,48 @@ To create a natural separation between options, simply use a `<hr>` element.
...
+      <auro-menuoption value="cars">Cars</auro-menuoption>
+      <auro-menuoption value="trucks">Trucks</auro-menuoption>
+      <auro-menuoption value="boats">Boats</auro-menuoption>
+      <auro-menuoption value="planes">Planes</auro-menuoption>
+      <auro-menuoption value="motorcycles">Motorcycles</auro-menuoption>
+    </auro-menu>
+  </auro-menu>
+  <!-- AURO-GENERATED-CONTENT:END -->
+</div>
+
```
Comment (blackfalcon) [inline demo/demo.md:134]: Example is missing the accordion code view

File: src/auro-menu.js [hunk 2]
```diff
@@ -185,15 +185,53 @@ class AuroMenu extends LitElement {
     }
   }
 
+  /**
+   * Used for applying indentation to each level of nested menu.
+   * @private
+   */
+  handleNestedMenus(menu) {
+    const nestedMenus = menu.querySelectorAll(':scope auro-menu');
```
Comment (blackfalcon) [inline src/auro-menu.js]: Removing `:scope` returns the same node list, not sure this is needed here.

File: src/auro-menu.js [hunk 3]
```diff
@@ -185,15 +185,53 @@ class AuroMenu extends LitElement {
...
+
+    if (nestedMenus.length === 0) return;
+
+    nestedMenus.forEach((nestedMenu) => {
+      const options = nestedMenu.querySelectorAll(':scope > auro-menuoption');
+
+      options.forEach((option) => {
+        option.innerHTML = '<span class="nestingSpacer"></span>' + option.innerHTML;
```
Comment (blackfalcon) [inline src/auro-menu.js:201]: This brings back horrible memories of using spacer gifs...

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: src/auro-menu.js (hunk 3 above)
Comment (jason-capsule42) [reply src/auro-menu.js:201]: I get that. I thought about putting padding to the left of each nested menu but that breaks the hover/active styles which would no longer be the full width of the menu.

I don't have another solution to cause progressive indenting with each layer. Do you have a different solution we can use?

**Review by blackfalcon — APPROVED** [review]

**Pull request merged**